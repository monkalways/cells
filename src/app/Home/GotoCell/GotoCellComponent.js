import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Button from '@material-ui/core/Button';
import FormControl from '@material-ui/core/FormControl';
import Input from '@material-ui/core/Input';
import InputLabel from '@material-ui/core/InputLabel';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';
import withStyles from '@material-ui/core/styles/withStyles';
import { withRouter } from 'react-router-dom';
import { compose } from 'recompose';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  }).isRequired,
};

export class GotoCellComponent extends Component {
  state = {
    cellName: '',
  };

  handleClick = (event) => {
    event.preventDefault();
    const { history } = this.props;
    const { cellName } = this.state;
    if (cellName) history.push(`/cells/${cellName}/`);
  };

  handleChange = (event) => {
    this.setState({
      cellName: event.target.value,
    });
  };

  render() {
    const { classes } = this.props;
    const { cellName } = this.state;
    return (
      <Paper className={classes.paper}>
        <Typography component="h1" variant="h5">
          Go to Cell Room?
        </Typography>
        <form className={classes.form}>
          <FormControl margin="normal" required fullWidth>
            <InputLabel htmlFor="cellName">Cell Name</InputLabel>
            <Input
              id="cellName"
              name="cellName"
              autoComplete="cellName"
              value={cellName}
              onChange={this.handleChange}
            />
          </FormControl>
          <Button
            type="submit"
            fullWidth
            variant="contained"
            color="primary"
            className={classes.submit}
            onClick={this.handleClick}
          >
            Go
          </Button>
        </form>
      </Paper>
    );
  }
}

GotoCellComponent.propTypes = propTypes;

export default compose(
  withStyles((theme) => ({
    paper: {
      marginTop: theme.spacing.unit * 8,
      marginLeft: theme.spacing.unit * 2,
      display: 'flex',
      flexDirection: 'column',
      alignItems: 'center',
      padding: `${theme.spacing.unit * 2}px ${theme.spacing.unit * 3}px ${theme
        .spacing.unit * 3}px`,
      height: theme.spacing.unit * 27,
      width: theme.spacing.unit * 40,
    },
    form: {
      width: '100%', // Fix IE 11 issue.
      marginTop: theme.spacing.unit,
    },
    submit: {
      marginTop: theme.spacing.unit * 3,
    },
  })),
  withRouter,
)(GotoCellComponent);
