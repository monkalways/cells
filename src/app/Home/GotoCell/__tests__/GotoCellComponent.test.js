import React from 'react';
import { shallow } from 'enzyme';

import { Button, Input } from '@material-ui/core';
import { GotoCellComponent } from '../GotoCellComponent';

describe('GotoCellComponent', () => {
  let classes;
  let push;

  beforeEach(() => {
    classes = {
      paper: 'paper',
      form: 'form',
      submit: 'submit',
    };
    push = jest.fn();
  });

  const setup = () => shallow(<GotoCellComponent classes={classes} history={{ push }} />);

  it('should render', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should set initial state', () => {
    const wrapper = setup();
    expect(wrapper.state().cellName).toEqual('');
  });

  it('should update usage state when input value changes', () => {
    const wrapper = setup();
    const event = {
      target: {
        value: 'c1',
      },
    };
    wrapper.find(Input).simulate('change', event);
    expect(wrapper.state().cellName).toEqual('c1');
  });

  it('should handle submit', () => {
    const wrapper = setup();
    wrapper.setState({ cellName: 'c1' });
    const preventDefault = jest.fn();
    wrapper.find(Button).simulate('click', { preventDefault });

    expect(preventDefault).toBeCalled();
    expect(push).toBeCalledWith('/cells/c1/');
  });
});
