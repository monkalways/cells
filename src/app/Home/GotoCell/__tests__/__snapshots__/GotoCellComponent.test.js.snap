// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GotoCellComponent should render 1`] = `
<WithStyles(Paper)
  className="paper"
>
  <WithStyles(Typography)
    component="h1"
    variant="h5"
  >
    Go to Cell Room?
  </WithStyles(Typography)>
  <form
    className="form"
  >
    <WithStyles(FormControl)
      fullWidth={true}
      margin="normal"
      required={true}
    >
      <WithStyles(InputLabel)
        htmlFor="cellName"
      >
        Cell Name
      </WithStyles(InputLabel)>
      <WithStyles(Input)
        autoComplete="cellName"
        id="cellName"
        name="cellName"
        onChange={[Function]}
        value=""
      />
    </WithStyles(FormControl)>
    <WithStyles(Button)
      className="submit"
      color="primary"
      fullWidth={true}
      onClick={[Function]}
      type="submit"
      variant="contained"
    >
      Go
    </WithStyles(Button)>
  </form>
</WithStyles(Paper)>
`;
